runtime/src/topo/hwloc/topo-hwloc.c

```diff
@@ -144,12 +158,18 @@ void chpl_topo_init(void) {
   // Allocate and initialize topology object.
   //
   CHK_ERR_ERRNO(hwloc_topology_init(&topology) == 0);
+  CHK_ERR_ERRNO(hwloc_topology_init(&wholeTopology) == 0);
 
   //
   // Perform the topology detection.
   //
   CHK_ERR_ERRNO(hwloc_topology_load(topology) == 0);
 
+  // See what we are missing.
+  int flags = HWLOC_TOPOLOGY_FLAG_WHOLE_SYSTEM;
+  CHK_ERR_ERRNO(hwloc_topology_set_flags(wholeTopology, flags) == 0);
+  CHK_ERR_ERRNO(hwloc_topology_load(wholeTopology) == 0);
```

Comment:
Loading a topology is relatively expensive (not really in the context of distributed set up, but at least for comm=none it adds a non-trivial amount of time to load a topology.) This is why we go out of our way to give qthreads our topology instead of having qthreads also load. For now I think we were talking about conditionalizing this on comm!=none

For more info on hwloc load time see https://github.com/chapel-lang/chapel/issues/7534 and https://github.com/Qthreads/qthreads/issues/41 and

Reply:
I modified the implementation to use the `online_cpuset` of the topology rather than getting the topology for the whole system.